### PR TITLE
feat(FX-3202): add analytic events for saved search M2

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3202]

#### Criteria:
* `tappedCreateAlert` → initial tap to record the interest/intent
* `editedSavedSearch` → when the edit action is performed
* `deletedSavedSearch` → when the delete action is performed
* `SavedSearchEdit` → User views a Saved Search Edit screen
* `SavedSearchList` → User views the Saved Search List view

Add support for the following events:
* `editedSavedSearch`
* `deletedSavedSearch`
* `tappedCreateAlert`
* `SavedSearchEdit`
* `SavedSearchList`

### Demo
Screenshot 2021-08-18 at 14 01 49

Screenshot 2021-08-19 at 16 34 53

Screenshot 2021-08-19 at 16 35 07

Screenshot 2021-08-20 at 19 07 52

Screenshot 2021-08-20 at 19 43 34

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add analytics for saved search M2 - dzmitry tratsiak

</details>

[FX-3202]: https://artsyproduct.atlassian.net/browse/FX-3202

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against debc721f1445191427f459eaa2b0d82cfae00d63
</p>

**Review by dblandin — APPROVED** [review]
Looks good! 👍🏼

File: src/lib/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx [hunk 1]
```diff
@@ -93,8 +95,21 @@ export const SavedSearchesList: React.FC<SavedSearchesListProps> = (props) => {
   )
 }
 
+export const SavedSearchesListWrapper: React.FC<SavedSearchesListProps> = (props) => {
+  return (
+    <ProvideScreenTracking
+      info={{
+        context_screen: Schema.PageNames.SavedSearchList,
+        context_screen_owner_type: OwnerType.savedSearch,
```
Comment (dblandin) [inline src/lib/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx:103]: **suggestion**: Looking at other owner types listed [here](https://github.com/artsy/cohesion/blob/master/src/Schema/Values/OwnerType.ts), it seems like we usually define a separate owner type for list pages. For example (artists vs artist, viewingRoom vs viewingRooms). I think we might want to define a new `savedSearches` owner type which never has a corresponding ID in the payload.

**Review by dblandin — COMMENTED** [review]
@dimatretyak I think we want to make one small change before this goes out to users.

**Pull request merged**